datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -286,8 +286,7 @@ impl SchemaAdapter {
         let projected_schema = Arc::new(self.table_schema.clone().project(projections)?);
 
         // Necessary to handle empty batches
-        let mut options = RecordBatchOptions::default();
-        options.row_count = Some(batch.num_rows());
+        let options = RecordBatchOptions::new().with_row_count(Some(batch.num_rows()));
```

Comment:
Update to use the nice API @askoa  added in https://github.com/apache/arrow-rs/pull/2729